### ISSUE TITLE
[BUG - Release] dwmapi.dll is causing crashes in Palworld

### ISSUE THREAD

**Issue by Darena-Bryant** [UE4SS-RE/RE-UE4SS#393]
**Branch or Release**
version 3.0

**Game and Engine Version**
Palworld 
app ID:1623730
build ID:13390747

**Describe the bug**
I recently installed this along with two mods: Gameplay Tweak and Admin Engine. All was fine at first, then I began getting more and more frequent memory errors where it claimed the game attempted to access a location it wasn't allowed to, or that didn't exist. After checking my memory to be sure the issue wasn't hardware, I found a suggestion to remove dwmapi.dll from the folder. Doing this made the game return to a working state. After playing for a bit, I tested if reinstalling it would work, and for a while, it did. Then the game went back to having memory issues. Today, I uninstalled it again as a test, and once again, the game returned to functioning. It seems like this file is somehow making the game crash by attempting to access restricted or non-existent memory locations.

**Mods directory**

[Mods.zip](https://github.com/UE4SS-RE/RE-UE4SS/files/14315373/Mods.zip)

**To Reproduce**
1. Launch game
2. Game immediately crashes.

**Expected behavior**
I would hope to be able to play the game without crashing.

**Desktop (please complete the following information):**
 - OS: Win10

**Comment by ElCapitanEduardo** [comment]
Same issue here. I removed the file and the game launched but the mods stopped working.

**Comment by Buckminsterfullerene02** [comment]
dwmapi.dll is what injects UE4SS into the game. So removing the file is basically removing UE4SS. 

Does this issue still occur if you turn off all of your mods but keep dwmapi.dll (i.e. UE4SS is still injected)? Please download the latest release - 3.0.1 - and use that.

If it does, then there is indeed something wrong with UE4SS. If it does not, then it is an issue with one or more of your mods.

**Comment by SafetyTheFlameWolf** [comment]
![Pal Crash Reporter 2024-02-16 1_20_09 PM](https://github.com/UE4SS-RE/RE-UE4SS/assets/20743232/7d139b22-0854-4b74-b28a-8c5a47bfbfb5)
![Pal Crash Reporter 2024-02-16 1_20_16 PM](https://github.com/UE4SS-RE/RE-UE4SS/assets/20743232/37a06d07-a31d-4547-ae15-3240814f9ea4)
this is what I am getting for the crash report screen (I set all the mods in the mod.txt folder to 0 as I am assuming that turns them off)

**Comment by UE4SS** [comment]
> I set all the mods in the mod.txt folder to 0 as I am assuming that turns them off

Setting a mod to to 0 in mods.txt doesn't turn off the mod if there's an `enabled.txt` file in that mods directory.
To turn off a mod, the mod must be set to 0, or not exist in mods.txt at all and there needs to be no `enabled.txt` file.

**Comment by SafetyTheFlameWolf** [comment]
> > I set all the mods in the mod.txt folder to 0 as I am assuming that turns them off
> 
> Setting a mod to to 0 in mods.txt doesn't turn off the mod if there's an `enabled.txt` file in that mods directory. To turn off a mod, the mod must be set to 0, or not exist in mods.txt at all and there needs to be no `enabled.txt` file.

ah ok. Ya getting rid of everything in th mods.txt seems to allow me to load into a game now. 

Posting screenshots of how everything looks like

![Win64 2024-02-16 1_37_48 PM](https://github.com/UE4SS-RE/RE-UE4SS/assets/20743232/4548a15a-ad17-4350-be64-9021a27705d5)
![mods txt - Notepad 2024-02-16 1_37_23 PM](https://github.com/UE4SS-RE/RE-UE4SS/assets/20743232/f04a93a2-9b2d-466c-a2ac-02115d96310c)
![BUG - Release  dwmapi dll is causing crashes in Palworld · Issue #393 · UE4SS-RE_RE-UE4SS - Opera 2024-02-16 1_37_12 PM](https://github.com/UE4SS-RE/RE-UE4SS/assets/20743232/e60e70f7-fa56-45f5-99d9-4c881045e4af)

**Comment by SafetyTheFlameWolf** [comment]
Ok so downloaded a mod and had it enabled and its working perfectly fine, so I'm guessing the issue is with one of the mods that came with the Mods folder

**Comment by Darena-Bryant** [comment]
> dwmapi.dll is what injects UE4SS into the game. So removing the file is basically removing UE4SS.
> 
> Does this issue still occur if you turn off all of your mods but keep dwmapi.dll (i.e. UE4SS is still injected)? Please download the latest release - 3.0.1 - and use that.
> 
> If it does, then there is indeed something wrong with UE4SS. If it does not, then it is an issue with one or more of your mods.

I'll give this a try soon. I only just saw this comment.

**Comment by Darena-Bryant** [comment]
Ok, so installed 3.0.1, cleared my mod file, updated the mod text file and reinstalled the mods I wanted to use. Am going to play a bit to see if everything is working properly or not. Last time I removed dwmapi.dll then restored it, the game ran well for a few hours, then began crashing constantly. Will update after a while of playing.

**Comment by Darena-Bryant** [comment]
I played for about two hours without any issues. I will try one more time before closing this though; I did the same thing before opening this, playing for a few hours, only for the game to start having errors at the end.

It turns out I had a large number of unknown files in my mod folder. I did not install these myself, and I think I picked them up by playing on a friend's server; I plan to ask them about what mods they were using. I cleaned everything up though, and hopefully this will have fixed the problem permanently. I'll make one more post sometime tomorrow, and if all is well, I'll close this.
